tests/integration_tests/test_automl.py

```diff
@@ -49,6 +51,40 @@ def to_name_set(features: List[Dict[str, Any]]) -> Set[str]:
     assert to_name_set(config[OUTPUT_FEATURES]) == to_name_set(output_features)
 
 
+@pytest.mark.distributed
+@pytest.mark.parametrize(
+    ("split_expected", "class_probs"),
+    [
+        pytest.param(False, np.array([0.33, 0.33, 0.34]), id="balanced"),
+        pytest.param(True, np.array([0.6, 0.2, 0.2]), id="imbalanced"),
+    ],
+)
+def test_autoconfig_preprocessing(split_expected, class_probs, ray_cluster_2cpu):
+    nrows = 1000
+    thresholds = np.cumsum((class_probs * nrows).astype(int))
+
+    df = pd.DataFrame(np.random.randint(0, 100, size=(nrows, 3)), columns=["A", "B", "C"])
+
+    def get_category(v):
+        if v < thresholds[0]:
+            return 0
+        if thresholds[0] <= v < thresholds[1]:
+            return 1
+        return 2
+
+    df["category"] = df.index.map(get_category).astype(np.int8)
+
+    config = create_auto_config(dataset=df, target="category", time_limit_s=1, tune_for_memory=False)
+
+    if split_expected:
+        assert PREPROCESSING in config
+        assert SPLIT in config[PREPROCESSING]
+        assert config[PREPROCESSING][SPLIT][TYPE] == "stratify"
+        assert config[PREPROCESSING][SPLIT][COLUMN] == "category"
+    else:
+        assert PREPROCESSING not in config
```

Comment:
I would break this test up into two separate tests, with a couple of helpers for the test setup.

In general, it's good to avoid conditionals/logic in tests:
https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-logic-in-tests

```python
def get_sample_df(class_probs) -> pd.DataFrame:
    nrows = 1000
    thresholds = np.cumsum((class_probs * nrows).astype(int))

    df = pd.DataFrame(np.random.randint(0, 100, size=(nrows, 3)), columns=["A", "B", "C"])

    def get_category(v):
        if v < thresholds[0]:
            return 0
        if thresholds[0] <= v < thresholds[1]:
            return 1
        return 2
    df["category"] = df.index.map(get_category).astype(np.int8)
    return df
```

```python
def test_autoconfig_preprocessing_balanced(ray_cluster_2cpu):
    df = get_sample_df([0.33, 0.33, 0.34])

    config = create_auto_config(dataset=df, target="category", time_limit_s=1, tune_for_memory=False)
    
    assert PREPROCESSING not in config

def test_autoconfig_preprocessing_imbalanced(ray_cluster_2cpu)
    df = get_sample_df([0.6, 0.2, 0.2])

    config = create_auto_config(dataset=df, target="category", time_limit_s=1, tune_for_memory=False)

    assert config[PREPROCESSING] == {...}
```